src/api/Feed.js

```diff
@@ -494,6 +494,50 @@ class Feed extends Base {
         });
     };
 
+    /**
+     * Updates a task in the new API
+     *
+     * @param {BoxItem} file - The file to which the task is assigned
+     * @param {string} task - The update task payload object
+     * @param {Function} successCallback - the function which will be called on success
+     * @param {Function} errorCallback - the function which will be called on error
+     * @return {void}
+     */
+    updateTaskNew = (
+        file: BoxItem,
+        task: TaskUpdatePayload,
+        successCallback: () => void = noop,
+        errorCallback: ErrorCallback = noop,
+    ) => {
+        if (!file.id) {
+            throw getBadItemError();
+        }
+
+        this.id = file.id;
+        this.errorCallback = errorCallback;
+        this.tasksNewAPI = new TasksNewAPI(this.options);
+        this.updateFeedItem({ isPending: true }, task.id);
+        this.tasksNewAPI.updateTask({
+            file,
+            task,
+            successCallback: (taskData: Task) => {
+                this.updateFeedItem(
+                    {
+                        ...taskData,
+                        isPending: false,
+                    },
+                    task.id,
+                );
+                if (!this.isDestroyed()) {
+                    successCallback();
```

Comment:
Should we pass the task back to the callback?

Reply:
We could, but it is not required at the moment. The form already receives latest data through props.